src/runtime-rs/crates/hypervisor/src/device/topology.rs

```diff
@@ -225,15 +304,38 @@ impl PCIeTopology {
             root_bus_devices: HashMap::with_capacity(PCIE_ROOT_BUS_SLOTS_CAPACITY as usize),
         };
 
+        // initialize port devices within PCIe Topology
+        let total_rp: usize = topo_config.device_info.pcie_root_port as usize;
+        let total_swp: usize = topo_config.device_info.pcie_switch_port as usize;
+
+        let mut devices_per_port: HashMap<PCIePort, Vec<TopologyPortDevice>> = HashMap::new();
+        if total_rp > 0 {
+            devices_per_port.insert(PCIePort::RootPort, Vec::with_capacity(total_rp));
+        }
+        if total_swp > 0 {
+            devices_per_port.insert(PCIePort::SwitchPort, Vec::with_capacity(total_swp));
+        }
+
         Some(Self {
             hypervisor_name: topo_config.hypervisor_name.to_owned(),
             root_complex,
             bridges: topo_config.device_info.default_bridges,
             pcie_root_ports: topo_config.device_info.pcie_root_port,
+            pcie_switch_ports: topo_config.device_info.pcie_switch_port,
             hotplug_vfio_on_root_bus: topo_config.device_info.hotplug_vfio_on_root_bus,
+            pcie_devices_per_port: devices_per_port,
         })
     }
 
+    pub fn get_pcie_port(&self) -> PCIePort {
```

Comment:
This function looks useful but I don't see it called from anywhere.  If that's the case actually, do we need it?